### PR TITLE
Change all indentation from 3 spaces to 2 using findent

### PR DESCRIPTION
**Pull request by JamieJQuinn**
Used [pip-installed findent 4.3.1](https://pypi.org/project/findent/) to change all indentation from 3 spaces to 2 using the following fish command:

``` fish
for f in (find . -name '*.f90'); findent -i2 -I2 < $f > $f.temp; mv $f.temp $f; end
```

### CODE REVIEW

File: src/solver.f90 [hunk 1]
```diff
@@ -272,8 +272,8 @@ subroutine divergence_v2p(self, div_u, u, v, w)
       class(field_t), intent(in) :: u, v, w
 
       class(field_t), pointer :: du_x, dv_x, dw_x, &
-                                 u_y, v_y, w_y, du_y, dv_y, dw_y, &
-                                 u_z, w_z, dw_z
```
Comment (semi-h) [inline src/solver.f90:276]: This is actually everywhere but had to comment on a specific example. I would expect the continuation line to start aligned with the first variable like it used to be.

File: src/solver.f90 [hunk 2]
```diff
@@ -283,11 +283,11 @@ subroutine divergence_v2p(self, div_u, u, v, w)
       ! Interpolation for v field in x
       ! Interpolation for w field in x
       call self%backend%tds_solve(du_x, u, self%xdirps, &
-                                  self%xdirps%stagder_v2p)
```
Comment (semi-h) [inline src/solver.f90:286]: Also here, I would expect the continuation from the parentheses, isn't this more common?

File: src/cuda/kernels/distributed.f90 [hunk 3]
```diff
@@ -1,653 +1,653 @@
...
-      b = blockIdx%x
-
-      ! store bulk coeffs in the registers
-      c_m4 = coeffs(1); c_m3 = coeffs(2); c_m2 = coeffs(3); c_m1 = coeffs(4)
-      c_j = coeffs(5)
-      c_p1 = coeffs(6); c_p2 = coeffs(7); c_p3 = coeffs(8); c_p4 = coeffs(9)
-      last_r = ffr(1)
-
-      du(i, 1, b) = coeffs_s(1, 1)*u_s(i, 1, b) &
-                    + coeffs_s(2, 1)*u_s(i, 2, b) &
```
Comment (semi-h) [inline src/cuda/kernels/distributed.f90:42]: I think this would be very confusing, very surprised fprettify allows this to be honest.

File: src/allocator.f90 [hunk 4]
```diff
@@ -1,37 +1,37 @@
-module m_allocator
```
Comment (semi-h) [inline src/allocator.f90:1]: This is also interesting, even the module declaration starts with two spacing. Is there a way to disable it? seems like a waste of two spaces to me.

File: src/allocator.f90 [hunk 5]
```diff
@@ -1,37 +1,37 @@
...
-     !! is empty (i.e. all initially allocated blocks are currently
-     !! referenced to) then a new block is allocated before a pointer
-     !! to it is returned.
-     !!
-     !! In order to reuse memory it is important that user code
-     !! release blocks when they are not needed anymore.  This is done
-     !! by calling the type bound procedure
-     !! [[m_allocator(module):release_block(subroutine)]].  The
-     !! released block is then pushed in front of the block list.
+  module m_allocator
```
Comment (semi-h) [inline src/allocator.f90:1]: Commented on the wrong side and its actually not visible due to that. I mean the 2 spaces before `module m_allocator`

File: src/solver.f90 (hunk 2 above)
Comment (Nanoseb) [reply src/solver.f90:286]: I believe we can play with this parameter
```
       --align_paren[=<n>]
              align continuation lines not preceded
                by '&' with preceding unmatched left parenthesis.
               n=0, or flag omitted: do not align.
               n=1: do align (default).

```
though the default should be do leave it as is

File: src/cuda/kernels/distributed.f90 (hunk 3 above)
Comment (semi-h) [reply src/cuda/kernels/distributed.f90:42]: Again not visible here on the comment tab, but `+ coeff...` starts from all the way left with two spaces, however I would expect it to start from right below the previous variable after the `=`

File: src/solver.f90 [hunk 6]
```diff
@@ -272,8 +272,8 @@ subroutine divergence_v2p(self, div_u, u, v, w)
       class(field_t), intent(in) :: u, v, w
 
       class(field_t), pointer :: du_x, dv_x, dw_x, &
-                                 u_y, v_y, w_y, du_y, dv_y, dw_y, &
-                                 u_z, w_z, dw_z
+        u_y, v_y, w_y, du_y, dv_y, dw_y, &
+        u_z, w_z, dw_z
```
Comment (semi-h) [inline src/solver.f90:276]: This is actually everywhere but had to comment on a specific example. I would expect the continuation line to start aligned with the first variable like it used to be.

File: src/cuda/kernels/distributed.f90 [hunk 7]
```diff
@@ -1,653 +1,653 @@
...
+    b = blockIdx%x
+
+    ! store bulk coeffs in the registers
+    c_m4 = coeffs(1); c_m3 = coeffs(2); c_m2 = coeffs(3); c_m1 = coeffs(4)
+    c_j = coeffs(5)
+    c_p1 = coeffs(6); c_p2 = coeffs(7); c_p3 = coeffs(8); c_p4 = coeffs(9)
+    last_r = ffr(1)
+
+    du(i, 1, b) = coeffs_s(1, 1)*u_s(i, 1, b) &
+      + coeffs_s(2, 1)*u_s(i, 2, b) &
```
Comment (semi-h) [inline src/cuda/kernels/distributed.f90:42]: Commenting here just to show the actual change with spacing

File: src/ordering.f90 [hunk 8]
```diff
@@ -69,30 +69,30 @@ pure subroutine get_index_reordering(out_i, out_j, out_k, in_i, in_j, in_k, reor
...
-         case (RDR_Y2Z)
-            dir_in = DIR_Y
-            dir_out = DIR_Z
-         case (RDR_Z2X)
-            dir_in = DIR_Z
-            dir_out = DIR_X
-         case (RDR_Z2Y)
-            dir_in = DIR_Z
-            dir_out = DIR_Y
+       case (RDR_X2Y)
```
Comment (semi-h) [inline src/ordering.f90:72]: Another interesting point. `case` is shifted only one space, not two, and also the lines inside the `case` are shifted by one.

File: src/poisson_fft.f90 [hunk 9]
```diff
@@ -110,89 +110,89 @@ subroutine waves_set(self, xdirps, ydirps, zdirps, sz)
...
+
+        yky(i) = cmplx(1._dp, 1._dp, kind=dp)*(ny*wp/ydirps%L)
+        eys(i) = cmplx(1._dp, 1._dp, kind=dp)*(ny*w/ydirps%L)
+        yk2(i) = cmplx(1._dp, 1._dp, kind=dp)*(ny*wp/ydirps%L)**2
       end do
       do i = ny/2 + 2, ny
-         yky(i) = yky(ny-i+2)
-         eys(i) = eys(ny-i+2)
-         yk2(i) = yk2(ny-i+2)
+        yky(i) = yky(ny-i+2)
```
Comment (semi-h) [inline src/poisson_fft.f90:140]: Looks like findent (at least with current settings) doesn't enforce anything between operators. (I think this particular line is not compatible with our current preference (spacing around `-` and `+` but no spacing around `*` and `/`) it would be nice if findent can identify and enforce a spacing based on our preference.

Comment (Nanoseb) [reply src/poisson_fft.f90:140]: findent doesn't support modifying these things. It is really mainly doing indentation.

File: src/common.f90 [hunk 10]
```diff
@@ -1,23 +1,23 @@
...
+    integer, parameter :: DIR_X = 1, DIR_Y = 2, DIR_Z = 3, DIR_C = 4
+    integer, parameter :: POISSON_SOLVER_FFT = 0, POISSON_SOLVER_CG = 1
 
-   integer, protected :: &
+    integer, protected :: &
       rdr_map(4, 4) = reshape([0, RDR_X2Y, RDR_X2Z, RDR_X2C, &
-                               RDR_Y2X, 0, RDR_Y2Z, RDR_Y2C, &
-                               RDR_Z2X, RDR_Z2Y, 0, RDR_Z2C, &
-                               RDR_C2X, RDR_C2Y, RDR_C2Z, 0], shape=[4, 4])
+      RDR_Y2X, 0, RDR_Y2Z, RDR_Y2C, &
```
[the remark this answers is not in the document]
Comment (Nanoseb) [reply src/common.f90:16]: To my surprise, it does work:

```
  integer, protected :: &
      rdr_map(4, 4) = reshape([0, RDR_X2Y, RDR_X2Z, RDR_X2C, &
                               RDR_Y2X, 0, RDR_Y2Z, RDR_Y2C, &
                               RDR_Z2X, RDR_Z2Y, 0, RDR_Z2C, &
                               RDR_C2X, RDR_C2Y, RDR_C2Z, 0], shape=[4, 4])

```
With `findent --align_paren=1 < common.f90`

File: src/poisson_fft.f90 (hunk 9 above)
Comment (semi-h) [reply src/poisson_fft.f90:140]: Its actually good. Because fprettify is too restrictive with spacing around operators, and I don't really like it. For example, when passing a keyword argument to a function, I would rather not using spaces around `=`, but of course prefer spaces around assingment `=` normally. And fprettify can't tell them apart and want you to pick one and use it everywhere. Also with `if` and `allocate`. You may want space after `if`, but no space after `allocate`, but fprettify won't let you do this because it considers these two in the same group and the rule is defined for the whole group...

File: src/common.f90 (hunk 10 above)
[the remark this answers is not in the document]
Comment (semi-h) [reply src/common.f90:16]: Great!

**Comment by Nanoseb** [comment]
> Used [pip-installed findent 4.3.1](https://pypi.org/project/findent/) to change all indentation from 3 spaces to 2 using the following fish command:
> 
> ```fish
> for f in (find . -name '*.f90'); findent -i2 -I2 < $f > $f.temp; mv $f.temp $f; end
> ```

I would add `--align_paren=1` (to align multilines parenthesis) and `--refactor_end` (to add subroutine names after `end subroutine`)

**Comment by semi-h** [comment]
Is there any setting for alignment for multi-line assignments and for multi-line variable declarations?

File: src/cuda/kernels/distributed.f90 (hunk 3 above)
Comment (Nanoseb) [reply src/cuda/kernels/distributed.f90:42]: yeah, I didn't see any way in the manual to indent continuation lines after a `=` nor a `::` with findent

File: src/allocator.f90 (hunk 5 above)
Comment (Nanoseb) [reply src/allocator.f90:1]: see https://github.com/xcompact3d/x3d2/pull/80#discussion_r1539124115

**Comment by Nanoseb** [comment]
> Is there any setting for alignment for multi-line assignments and for multi-line variable declarations?

no, unfortunately I haven't seen anything about that in the help page

File: src/cuda/kernels/distributed.f90 (hunk 3 above)
Comment (semi-h) [reply src/cuda/kernels/distributed.f90:42]: with `::` its not the end of the world, one can argue its better to follow findent way, but with `+` I think things would get very confusing and hard to follow.

**Comment by Nanoseb** [comment]
Because the main limitation of findent is line continuation after `=`, `::` and `:`. I wrote a small script to handle only these. It is to be used on findent output.

```
findent < in.f90 | script/indent_continuation.py > out.f90
```

**Comment by Nanoseb** [comment]
using fprettify instead see #81
